README.md

```diff
@@ -0,0 +1,79 @@
+# Geneve
+
+Geneve is a data generation tool, its name stands for GENerate EVEnts.
+
+To better understand its basics, consider the Elastic Security's
+[detection engine](https://www.elastic.co/guide/en/security/current/detection-engine-overview.html).
+It regularly searches one or more indices for suspicious events, when a
+match is found it creates an alert. To do so it needs detection rules
+which define what a _suspicious event_ looks like.
+
+The original goal of Geneve is then summarized by:
+
+> Given a detection rule, generate source events that would trigger an alert creation.
+
+It does so by analyzing the rule, building an abstract syntax tree of the
+enclosed query and translating it to an intermediate language that is used
+for generating documents (= events) over and over.
+
+What became obvious over time is that the query at the heart of each rule
+is actually a powerful way to drive the documents generation that goes
+well beyond the alerts triggering.
+
+Additionally, one thing is generating garbage data that satisfies a rule
+and another is generating realistic data that can be analyzed with Kibana,
+which is an implicit goal of the tool.
+
+This last is a quite harder nut to crack than the original goal and is
+currently under development.
+
+If you want to try it, read [Getting started](docs/getting_started.md).
+
+# Status
+
+## Data modeling
+
+The rules/queries parsing, AST creation and IR generation are quite
+developed and rigorously tested by the CI/CD pipelines. The generated
+events are good enough to trigger many of the expected alerts on various
+versions of the stack, from 8.2.0 to 8.6.0, but the work is necessarily
+incomplete albeit as correct as possible.
+
+The detection rules set used for the tests is separately loaded into
+Geneve and is currently locked to version 8.2.0 (718 rules in total). Next
+step is to use the rules preloaded in the Kibana under test
+(https://github.com/elastic/geneve/issues/125).
```

Comment:
I've added a comment to https://github.com/elastic/geneve/issues/125 with some details, but just wanted to note that as of https://github.com/elastic/kibana/issues/139926 in `8.7` we're no longer loading prebuilt rules as part of assets bundled with Kibana, but rather as a bundled fleet package. This will only change things if you were planning on fetching them directly from the filesystem instead of from the API. Let me know if you have any questions when the time comes to work this though, happy to help here 🙂

Reply:
Thanks, let's continue there.